src/openrct2/peep/Guest.cpp

```diff
@@ -2487,11 +2487,18 @@ static Vehicle* peep_choose_car_from_ride(Peep* peep, Ride* ride, std::vector<ui
 
     peep->CurrentCar = car_array[chosen_car];
 
-    Vehicle* vehicle = GET_VEHICLE(ride->vehicles[peep->CurrentTrain]);
-
+    Vehicle* vehicle = GetEntity<Vehicle>(ride->vehicles[peep->CurrentTrain]);
+    if (vehicle == nullptr)
+    {
+        return nullptr;
+    }
     for (int32_t i = peep->CurrentCar; i > 0; --i)
     {
-        vehicle = GET_VEHICLE(vehicle->next_vehicle_on_train);
+        vehicle = GetEntity<Vehicle>(vehicle->next_vehicle_on_train);
+        if (vehicle == nullptr)
+        {
+            return nullptr;
+        }
```

Comment:
I wonder if it's clearer if we do:
```c++
    auto vehicle = GetEntity<Vehicle>(ride->vehicles[peep->CurrentTrain]);
    for (int32_t i = peep->CurrentCar; i > 0 && vehicle != nullptr; --i)
    {
        vehicle = GetEntity<Vehicle>(vehicle->next_vehicle_on_train);
    }
    return vehicle;
```

Reply:
There's actually a function for replacing all of this. I'm leaving that for a later refactor though.